apps/gdalalg_vector_change_field_type.cpp

```diff
@@ -0,0 +1,244 @@
+
+#include "gdalalg_vector_change_field_type.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALVectorChangeFieldTypeAlgorithm::GDALVectorChangeFieldTypeAlgorithm()          */
+/************************************************************************/
+
+GDALVectorChangeFieldTypeAlgorithm::GDALVectorChangeFieldTypeAlgorithm(
+    bool standaloneStep)
+    : GDALVectorPipelineStepAlgorithm(NAME, DESCRIPTION, HELP_URL,
+                                      standaloneStep)
+{
+    AddActiveLayerArg(&m_activeLayer);
+    auto &arg = AddFieldNameArg(&m_fieldName).SetRequired();
+    arg.SetAutoCompleteFunction(
+        [this, &arg](const std::string &)
+        {
+            std::vector<std::string> ret;
+            if (!this->GetInputDatasets().empty())
+            {
+                // Input layer
+                auto poDS = this->GetInputDatasets()[0].GetDatasetRef();
+                if (poDS)
+                {
+                    OGRLayer *poLayer = nullptr;
+                    if (!m_activeLayer.empty())
+                        poLayer =
+                            const_cast<GDALDataset *>(poDS)->GetLayerByName(
+                                m_activeLayer.c_str());
+                    else
+                        poLayer = const_cast<OGRLayer *>(poDS->GetLayer(0));
+                    if (poLayer)
+                    {
+                        auto poDefn = poLayer->GetLayerDefn();
+                        const int nFieldCount = poDefn->GetFieldCount();
+                        for (int iField = 0; iField < nFieldCount; iField++)
+                            ret.push_back(
+                                poDefn->GetFieldDefn(iField)->GetNameRef());
+                    }
+                }
+                return ret;
+            }
+        });
+    AddFieldTypeSubtypeArg(&m_newFieldType, &m_newFieldSubType,
+                           &m_newFieldTypeSubTypeStr)
+        .SetRequired();
+    AddValidationAction(
+        [this]
+        {
+            auto mInDS = m_inputDataset[0].GetDatasetRef();
+            auto layer = m_activeLayer.empty()
+                             ? mInDS->GetLayer(0)
+                             : mInDS->GetLayerByName(m_activeLayer.c_str());
+            if (!layer)
+            {
+                CPLError(CE_Failure, CPLE_AppDefined, "Cannot find layer '%s'",
+                         m_activeLayer.c_str());
+                return false;
+            }
+            if (layer->GetLayerDefn()->GetFieldIndex(m_fieldName.c_str()) < 0)
+            {
+                CPLError(CE_Failure, CPLE_AppDefined,
+                         "Cannot find field '%s' in layer '%s'",
+                         m_fieldName.c_str(), layer->GetName());
+                return false;
+            }
+            return true;
+        });
+}
+
```

Comment:
I know this is a very special case but some user has invented it some time ago not me. What would happen with this CSV?

```
tricky.csv
======
foo,foo,foo
a,b,c

gdal vector info tricky.csv --of text
INFO: Open of `tricky.csv'
      using driver `CSV' successful.

Layer name: tricky
Geometry: None
Feature Count: 1
Layer SRS WKT:
(unknown)
foo: String (0.0)
foo: String (0.0)
foo: String (0.0)

```
I do not really care, I think such CSV should not exist.